### PR TITLE
Feature/copy update pricingrules

### PR DESCRIPTION
**Pull request by eguzki**
`3scale copy service` and `3scale update service` commands add pricing rules to the list of entities to copy/update.

Needs new release of 3scale-api-ruby with the following PR merged
https://github.com/3scale/3scale-api-ruby/pull/47

### CODE REVIEW

**Comment by eguzki** [comment]
ready for review @mikz @miguelsorianod

File: lib/3scale_toolbox/commands/copy_command/copy_service.rb [hunk 1]
```diff
@@ -40,7 +40,8 @@ def run
             Tasks::CopyApplicationPlansTask.new(context),
             Tasks::CopyLimitsTask.new(context),
             Tasks::DestroyMappingRulesTask.new(context),
-            Tasks::CopyMappingRulesTask.new(context)
+            Tasks::CopyMappingRulesTask.new(context),
+            Tasks::CopyPricingRulesTask.new(context)
```
Comment (mikz) [inline lib/3scale_toolbox/commands/copy_command/copy_service.rb]: The same comment as 
https://github.com/3scale/3scale_toolbox/pull/101#discussion_r263277216

File: lib/3scale_toolbox/tasks/copy_pricingrules_task.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  module Tasks
+    class CopyPricingRulesTask
+      include CopyTask
+      include Helper
+
+      def call
+        metrics_map = metrics_mapping(source.metrics, target.metrics)
+        plan_mapping = application_plan_mapping(source.plans, target.plans)
+        plan_mapping.each do |plan_id, target_plan|
+          pricingrules_source = source.remote.list_pricingrules_per_application_plan(plan_id)
```
Comment (mikz) [inline lib/3scale_toolbox/tasks/copy_pricingrules_task.rb]: Why is PricingRule defined as `pricingrule` but ApplicationPlan as `application_plan` ?

File: lib/3scale_toolbox/tasks/copy_pricingrules_task.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        end
+      end
+
+      def parse_metric_id_from_links(pricing_rule_links)
+        # This method will not be necessary when https://github.com/3scale/porta/issues/671 is fixed.
+        # When method
+        # https://3scale-supertest-admin.3scale.net/admin/api/services/2555417769646/metrics/2555418155744/methods/2555418155754
+        # When metric
+        # https://3scale-supertest-admin.3scale.net/admin/api/services/2555417769646/metrics/2555418160039
+        metric_link = pricing_rule_links.select { |link| link['rel'] == 'metric' }[0]
```
Comment (mikz) [inline lib/3scale_toolbox/tasks/copy_pricingrules_task.rb]: Might be worth waiting for https://github.com/3scale/porta/pull/688 ?

File: lib/3scale_toolbox/commands/copy_command/copy_service.rb (hunk 1 above)
Comment (eguzki) [reply lib/3scale_toolbox/commands/copy_command/copy_service.rb]: :+1:

File: lib/3scale_toolbox/tasks/copy_pricingrules_task.rb (hunk 2 above)
Comment (eguzki) [reply lib/3scale_toolbox/tasks/copy_pricingrules_task.rb]: :man_shrugging: 

I will change

File: lib/3scale_toolbox/tasks/copy_pricingrules_task.rb (hunk 3 above)
Comment (eguzki) [reply lib/3scale_toolbox/tasks/copy_pricingrules_task.rb]: we will wait for it

**Comment by eguzki** [comment]
@mikz all required changes applied

**Comment by mikz** [comment]
One nitpick regarding the git history. I think it is a good practice to commit code and tests in the same commit. The tests testing that code should be in the same commit because it is easier to bisect and quite obvious in the history that the tests belong to some code and they are green.

**Pull request merged**